nodewatcher/nodewatcher.py

```diff
@@ -236,7 +236,6 @@ def main():
                     else:
                         has_pending_jobs, error = _has_pending_jobs(scheduler_module)
                         if not error and not has_pending_jobs:
-                            os.remove(idletime_file)
                             try:
                                 _self_terminate(asg_name, asg_conn, instance_id)
```

Comment:
probably not related to this patch but just noticed and wanted to mention: _self_terminate might not terminate the instance if it became essential in the meanwhile but at the same time termination_in_progress is set to True right after meaning this instance will never terminate...

Reply:
Thank you for highlighting it. It could be part of a future patch.